Extensions/DialogueTree/dialoguetools.js

```diff
@@ -0,0 +1,583 @@
+  this.dialogueData = this.dialogue.next().value; //Breaks here if json resource is used
+  this.dialogueIsRunning = true;
+  gdjs.dialogueTree.advanceDialogue();
+};
+
+/**
+ * Internal methods to check the type of a Dialogue Line
+ */
+gdjs.dialogueTree._lineTypeIsText = function() {
+  return this.dialogueData instanceof bondage.TextResult;
+};
+gdjs.dialogueTree._lineTypeIsOptions = function() {
+  return this.dialogueData instanceof bondage.OptionsResult;
+};
+gdjs.dialogueTree._lineTypeIsCommand = function() {
+  return this.dialogueData instanceof bondage.CommandResult;
+};
+
+/**
+ * This is the main lifecycle function.It runs once only when the user is advancing the dialogue to the next line.
+ * Use this to progress Dialogue to the next line. Hook it to your game input.
+ * Note that this action can be influenced by any <<wait>> commands, but they work only if you have at least one isCommandCalled condition.
+ */
+gdjs.dialogueTree.advanceDialogue = function() {
+  if (this.pauseScrolling || !this.dialogueIsRunning) return;
+
+  this.optionsCount = 0;
+  this.selectedOption = -1;
+  this.selectedOptionUpdated = false;
+
+  if (gdjs.dialogueTree._lineTypeIsText()) {
+    if (
+      this.dialogueDataType === "options" ||
+      this.dialogueDataType === "text" ||
+      !this.dialogueDataType
+    ) {
+      this.clipTextEnd = 0;
+      this.dialogueText = this.dialogueData.text;
+      this.commandCalls = [];
+    } else {
+      this.dialogueText += this.dialogueData.text;
+    }
+
+    this.dialogueDataType = "text";
+    this.dialogueBranchTags = this.dialogueData.data.tags;
+    this.dialogueBranchTitle = this.dialogueData.data.title;
+    this.dialogueBranchBody = this.dialogueData.data.body;
+    this.dialogueData = this.dialogue.next().value;
+  } else if (gdjs.dialogueTree._lineTypeIsOptions()) {
+    this.dialogueDataType = "options";
+    this.optionsCount = this.dialogueData.options.length;
+    this.options = this.dialogueData.options;
+    this.selectedOptionUpdated = true;
+  } else if (gdjs.dialogueTree._lineTypeIsCommand()) {
+    this.dialogueDataType = "command";
+
+    var command = this.dialogueData.text.split(" ");
+    // if last command was to wait, increase time by one
+    var offsetTime =
+      this.commandCalls.length &&
+      this.commandCalls[this.commandCalls.length - 1].cmd === "wait"
+        ? 1
+        : 0;
+    this.commandCalls.push({
+      cmd: command[0],
+      params: command,
+      time: this.dialogueText.length + offsetTime
+    });
+    this.dialogueData = this.dialogue.next().value;
+    gdjs.dialogueTree.advanceDialogue();
+  } else {
+    this.dialogueDataType = "unknown";
+  }
+
+  if (gdjs.dialogueTree._lineTypeIsCommand()) {
+    this.dialogueDataType = "command";
+    gdjs.dialogueTree.advanceDialogue();
+  }
+
+  // dialogue has finished
+  if (!this.dialogueData) {
+    this.dialogueIsRunning = false;
+  }
+};
+
+/**
+ * use this to get the current Dialogue Tree branch title
```

Comment:
Can you remove all the "use this to" from the descriptions? Descriptions of JS functions should be straight to the point: `Get the current Dialogue Tree branch title`.

Reply:
nit: Don't forget capital letters at the beginning of a sentence